metron-platform/metron-parsers/src/test/java/org/apache/metron/parsers/asa/BasicAsaParserTest.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Test
+    public void testShortTimestamp() {
+        String rawMessage = "<174>Jan  5 14:52:35 10.22.8.212 %ASA-6-302015: Built inbound UDP connection 76245506 for outside:10.22.8.110/49886 (10.22.8.110/49886) to inside:192.111.72.8/8612 (192.111.72.8/8612) (user.name)";
+        JSONObject asaJson = asaParser.parse(rawMessage.getBytes()).get(0);
+        assertEquals(asaJson.get("original_string"), rawMessage);
+        assertTrue(asaJson.get("ip_src_addr").equals("10.22.8.110"));
+        assertTrue(asaJson.get("ip_dst_addr").equals("192.111.72.8"));
+        assertTrue(asaJson.get("ip_src_port").equals(new Integer(49886)));
+        assertTrue(asaJson.get("ip_dst_port").equals(new Integer(8612)));
+        assertTrue((long) asaJson.get("timestamp") == 1452005555000L);
+    }
+
+    @Test
+    public void testNoPatternForTag() {
+        String rawMessage = "<165>Aug 16 2016 04:08:36: %ASA-5-713049: Group = 172.22.136.20, IP = 172.22.136.20, Security negotiation complete for LAN-to-LAN Group (172.22.136.20)  Initiator, Inbound SPI = 0x891fb03f, Outbound SPI = 0xbe4b5d8d";
+        JSONObject asaJson = asaParser.parse(rawMessage.getBytes()).get(0);
+        assertEquals(asaJson.get("original_string"), rawMessage);
+        assertTrue((long) asaJson.get("timestamp") == 1471320516000L);
+    }
+
+    @Test
+    public void testInvalidIpAddr() {
+        String rawMessage = "<164>Aug 05 2016 01:01:34: %ASA-4-106023: Deny tcp src Inside:10.30.9.121/54580 dst Outside:192.168.256.51/42028 by access-group \"Inside_access_in\" [0x962df600, 0x0]";
+        JSONObject asaJson = asaParser.parse(rawMessage.getBytes()).get(0);
+        assertEquals(asaJson.get("original_string"), rawMessage);
+        assertTrue((long) asaJson.get("timestamp") == 1470358894000L);
+        assertNull(asaJson.get("ip_dst_addr"));
+    }
+
+    @Test
+    public void testIp6Addr() {
+        String rawMessage = "<174>Jan  5 14:52:35 10.22.8.212 %ASA-6-302015: Built inbound UDP connection 76245506 for outside:2001:db8:85a3::8a2e:370:7334/49886 (10.22.8.110/49886) to inside:2001:0db8:85a3:0000:0000:8a2e:0370:7334/8612 (192.111.72.8/8612) (user.name)";
+        JSONObject asaJson = asaParser.parse(rawMessage.getBytes()).get(0);
+        assertEquals(asaJson.get("original_string"), rawMessage);
+        assertTrue(asaJson.get("ip_src_addr").equals("2001:db8:85a3::8a2e:370:7334"));
+        assertTrue(asaJson.get("ip_dst_addr").equals("2001:0db8:85a3:0000:0000:8a2e:0370:7334"));
+        assertTrue(asaJson.get("ip_src_port").equals(new Integer(49886)));
+        assertTrue(asaJson.get("ip_dst_port").equals(new Integer(8612)));
+        assertTrue((long) asaJson.get("timestamp") == 1452005555000L);
+    }
+
+    @Test
+    public void testUnexpectedMessage() {
+        String rawMessage = "-- MARK --";
+        try {
+            JSONObject asaJson = asaParser.parse(rawMessage.getBytes()).get(0);
+        } catch (RuntimeException e) {
+            assertTrue(true);
```

Comment:
In this case, I'd prefer using JUnit's Rules to test this. RuntimeExceptions are fairly generic, and Rules would allow the specific exception message to be verified.

e.g.
```
TestThing testThing = new TestThing();
thrown.expect(NotFoundException.class);
thrown.expectMessage(startsWith("some Message"));
```

https://github.com/junit-team/junit4/wiki/exception-testing